### DISCUSSION TITLE
How to use Argo CD App metadata in notification template?

### DISCUSSION THREAD

**Discussion by toddradin** [argoproj/argo-rollouts#3990] [Q&A] [answered]
We are using Argo CD notifications currently and I am testing out Argo Rollouts and trying to implement the equivalent notifications for Argo Rollouts. For context, Argo CD and Argo Rollouts are currently installed in different namespaces on the same cluster.

In the docs, there are multiple [examples](https://argoproj.github.io/argo-rollouts/generated/notification-services/webhook/) of using `.app.metadata.*` in the templates. We use these in our Argo CD notifications and they work as expected. However, for Argo Rollouts, it seems the Rollout notification templates cannot read `.app` as nothing is displayed at all. 

Has anyone run into this before and figured out a way to get the Rollout notification template to be able to access the Argo Application data?

**Comment by kostis-codefresh** [comment]
The Rollout controller has no visibility into Argo CD or Git or anything that is not defined in a Rollout resources.

So the notifications only have access to the Rollout context.

**Reply by toddradin** [reply]
I see. I've also tried subscribing Argo CD Applications to the Rollouts triggers (e.g. `on-rollout-completed`), but I haven't been able to get that to work either. Is that even possible? 

If not, are the docs just incorrect? A lot of the [examples](https://argoproj.github.io/argo-rollouts/generated/notification-services/webhook/) make it appear that it can be done.

**Reply by kostis-codefresh** [reply, marked as answer]
Yes the docs are probably incorrect. Because the notification engine is common between the two projects at some point the notification docs were merged into both projects (Argo CD, Argo Rollouts). But I believe nobody has ever checked that the part that was added in Rollouts is actually documented to mention rollouts.

Could you please open a github issue for this, so that we can keep track of it?

In general however

Argo CD notifications -> only know what happens with Argo CD apps
Argo Rollouts notifications -> only know what happens with Rollout objects.

There is no mix between the two (at least in a default installation)

**Reply by toddradin** [reply]
Thank you for the information! Created https://github.com/argoproj/argo-rollouts/issues/3998 for this.
